src/app/qgisapp.cpp

```diff
@@ -7307,15 +7346,51 @@ void QgisApp::saveAsVectorFileGeneral( QgsVectorLayer *vlayer, bool symbologyOpt
   if ( !vlayer )
     return;
 
+  auto onSuccess = [this, vlayer]( const QString & newFilename,
+                                   bool addToCanvas,
+                                   const QString & layerName,
+                                   const QString & encoding,
+                                   const QString & vectorFileName )
+  {
+    if ( addToCanvas )
+    {
+      QString uri( newFilename );
+      if ( !layerName.isEmpty() )
+        uri += "|layername=" + layerName;
+      this->addVectorLayers( QStringList( uri ), encoding, QStringLiteral( "file" ) );
+    }
+    this->emit layerSavedAs( vlayer, vectorFileName );
+    this->messageBar()->pushMessage( tr( "Layer Exported" ),
+                                     tr( "Successfully saved vector layer to <a href=\"%1\">%2</a>" ).arg( QUrl::fromLocalFile( QFileInfo( newFilename ).path() ).toString(), QDir::toNativeSeparators( newFilename ) ),
```

Comment:
Also here, use file location instead of path location.

Reply:
Thanks, both fixed